packages/foundry/contracts/witch/WitchV2.sol

```diff
@@ -327,27 +359,50 @@ contract WitchV2 is AccessControl {
         // Update Cauldron and local auction data
         _updateAccounting(
             vaultId,
-            vault.ilkId,
+            auction_.ilkId,
             auction_.baseId,
             auction_,
             inkOut,
             artIn
         );
 
         // Move the assets
-        if (inkOut != 0) {
-            // Give collateral to the user
-            IJoin ilkJoin = ladle.joins(vault.ilkId);
-            require(ilkJoin != IJoin(address(0)), "Join not found");
-            ilkJoin.exit(to, inkOut.u128());
-        }
+        uint256 auctioneerCut = _payInk(
+            auction_.auctioneer,
+            inkOut,
+            auction_.ilkId,
+            to
+        );
         if (artIn != 0) {
-            // Burn fyToken from user
-            DataTypes.Series memory series = cauldron.series(vault.seriesId);
+            // Burn fyToken from liquidator
+            DataTypes.Series memory series = cauldron.series(auction_.seriesId);
             series.fyToken.burn(msg.sender, artIn);
         }
 
         _collateralBought(vaultId, to, inkOut, artIn);
+        // What the liquidator actually gets is the computed inkOut minus what's paid to the auctioneer
+        inkOut -= auctioneerCut;
+    }
+
+    function _payInk(
+        address auctioneer,
+        uint256 inkOut,
+        bytes6 ilkId,
+        address to
+    ) internal returns (uint256 auctioneerCut) {
+        if (inkOut != 0) {
+            IJoin ilkJoin = ladle.joins(ilkId);
+            require(ilkJoin != IJoin(address(0)), "Join not found");
+
+            if (auctioneer != to) {
+                // Pay auctioneer's cut
+                auctioneerCut = inkOut.wmul(auctioneerReward);
```

Comment:
I reckon `auctioneerCut` should be calculated in `_calcPayout`, and passed to `_payInk` as a parameter.